### PR TITLE
Add edit, delete to `keybase chat api`

### PR DESCRIPTION
**Pull request by patrickxb**
r? @mmaxim

### CODE REVIEW

File: go/service/chat_boxer.go [hunk 1]
```diff
@@ -84,13 +84,21 @@ func (b *chatBoxer) unboxMessageWithKey(ctx context.Context, msg chat1.MessageBo
...
-	packedBody, err := b.open(msg.BodyCiphertext, key)
-	if err != nil {
-		return chat1.MessagePlaintext{}, err
-	}
 	var body chat1.BodyPlaintext
-	if err := b.unmarshal(packedBody, &body); err != nil {
-		return chat1.MessagePlaintext{}, err
+	skipBodyVerification := false
+	if len(msg.BodyCiphertext.E) == 0 {
+		if msg.ServerHeader.SupersededBy == 0 {
```
Comment (mmaxim) [inline go/service/chat_boxer.go]: Doesn't this need to check the `SupersedBy` on MessageBoxed?

Comment (patrickxb) [inline go/service/chat_boxer.go]: Currently, that's always nil.

Comment (mmaxim) [inline go/service/chat_boxer.go]: Is there another ticket for the server integration for that?

Comment (patrickxb) [inline go/service/chat_boxer.go]: added 3863

**Pull request merged**